pkg/db/db_test.go

```diff
@@ -0,0 +1,109 @@
+package db
+
+import (
+	"database/sql"
+	"fmt"
+	"os"
+	"testing"
+)
+
+var (
+	testDB    *Database
+	testDBErr error
+)
+
+func TestMain(m *testing.M) {
+	setup()
+	c := m.Run()
+	teardown()
+	os.Exit(c)
+}
+
+func setup() {
+	conn, err := sql.Open("sqlite3", ":memory:")
+	testDB = (*Database)(conn)
+	testDBErr = err
+	sql := `
+	  CREATE TABLE IF NOT EXISTS migrations(
+		uid TEXT,
+		name TEXT,
+		host TEXT,
+		data_type TEXT,
+		data TEXT,
+		PRIMARY KEY(uid, name, host, data_type)
+	);
+	`
+	if err := testDB.Exec(sql); err != nil {
+		fmt.Fprintf(os.Stderr, "failed to create table: %v\n", err)
+		os.Exit(1)
+	}
+}
+
+func teardown() {
+	if err := testDB.Close(); err != nil {
+		fmt.Fprintf(os.Stderr, "failed to close DB connection: %v", err)
+		os.Exit(1)
+	}
+}
+
+func TestConnect(t *testing.T) {
+	if testDB == nil {
+		t.Errorf("Connection is nil")
+	}
+	if testDBErr != nil {
+		t.Errorf("Connection to the DB failed: %v", testDBErr)
```

Comment:
s/Errorf/Fatalf/g

`Fatalf` won't allow the execution of other tests. Since there is no a valid connection, it makes sense to fail the test right here.